### PR TITLE
Updated package setup

### PR DESCRIPTION
**Pull request by Goldziher**
This PR makes substantial improvements to the package setup - introducing pre-commit and making our lives easier. I suggest it is reviewed only after the huge PR from which it is branched is merged, to make this simpler.

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -1,8 +1,20 @@
-Copyright (c) Maxim Kurnikov.
-All rights reserved.
+Copyright (c) Maxim Kurnikov
```
Comment (sobolevn) [inline LICENSE.txt:1]: Let's keep the license file as-is, please.

File: release.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (sobolevn) [inline release.sh:1]: `#!/usr/bin/env bash`

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks! Let's rebase this one properly and I will review it once again. 👍

**Comment by Goldziher** [comment]
this PR was remade, closing this one.

**Pull request closed without merging**